CanonicalLayouts/list-detail-compose/app/src/main/java/com/example/listdetailcompose/ui/ListDetailSample.kt

```diff
@@ -222,13 +290,36 @@ private fun ListContent(
 private fun DetailContent(
     definedWord: DefinedWord?,
     modifier: Modifier = Modifier,
+    isSmallerThanExpanded: Boolean,
+    isDetailVisible: Boolean,
+    sharedTransitionScope: SharedTransitionScope,
+    animatedVisibilityScope: AnimatedVisibilityScope
 ) {
     Column(
         modifier = modifier
             .verticalScroll(rememberScrollState())
             .padding(vertical = 16.dp)
     ) {
         if (definedWord != null) {
+
+            val imageModifier = Modifier.padding(horizontal = 8.dp)
+            if (isSmallerThanExpanded && isDetailVisible) {
+                with(sharedTransitionScope) {
+                    val state = rememberSharedContentState(key = definedWord.word)
+                    imageModifier.then(
+                        Modifier.sharedElement(
+                            state,
+                            animatedVisibilityScope = animatedVisibilityScope
+                        )
+                    )
+                }
+            }
```

Comment:
This won't work, since the result of `Modifier.then` isn't used anywhere.

```suggestion
            val imageModifier = Modifier
                .padding(horizontal = 8.dp)
                .then(
                    if (isSmallerThanExpanded && isDetailVisible) {
                        with(sharedTransitionScope) {
                            val state = rememberSharedContentState(key = definedWord.word)
                            Modifier.sharedElement(
                                state,
                                animatedVisibilityScope = animatedVisibilityScope
                            )
                         }
                    } else {
                        Modifier
                    } 
                )
```